### ISSUE TITLE
Persistence update

### ISSUE THREAD

**Issue by mandolyte** [unfoldingWord/tc-create-app#185]
One improvement and one bug fix.
_Improvement:_ since `localforage` automatically stringifies/parses objects going in an out, there is no need for the code to do the same thing. This also means that that objects can be expanded and viewed normally in the browser console, which is nice.
_Bug:_ the `saveState` function tests the value for truth using this:`if (value)`. This means that the application cannot save a key with the value zero, since it will test false. Replace with `if ( value === null )`. NOTE: `localforage` uses null for invalid keys, never returns an undefined (for compatibility reasons with local/session storage mechanisms).

**Comment by mandolyte** [comment]
For quick reference:
https://localforage.github.io/localForage/#data-api-setitem

**Comment by elsylambert** [comment]
Verified in translationCore Create - v1.0.0-beta.7. Remember me in Login works fine. When the user checks the checkbox"Remember me" in the login form, app remembers the session and opens the last file accessed when the app is closed accidentally. 


If the option is not checked, if user accidentally closes the app, the app cannot open the last file accessed. the spinner keeps on spinning on the file page(blank) until the user closes the file. and then Login page shows up. Is this the expected behaviour? @klappy @mandolyte

**Comment by mandolyte** [comment]
@RoyalSix I verified @elsylambert 's finding. It seems as if:
a) no login credentials are persisted (this is expected since the remember checkbox is unchecked in this scenario)
b) but when the app is simply closed after getting to a file (I used utn for Ruth), and then app is launched a second time, then the stepper seems to advance to the the file, even though it has no credentials. I think this might explain the endless spinner.

If I close the resource from the tool bar, it proceeds to the login; from there it works.

Since I have some outstanding PRs that might impact this, perhaps we ought to wait until beta.8 is out?

@elsylambert I think this deserves a new issue, since I doubt this particular issue is related.
